src/main/java/Duke.java

```diff
@@ -119,14 +122,38 @@ private static void addTask(String promptDescription) throws DukeException {
     }
 
     // Marks a task as done and informs the user about it
-    private static void completeTask(int taskNo) throws DukeException {
-        if (taskNo > 0 & taskNo <= tasksList.size()) {
-            Task taskToComplete = tasksList.get(taskNo - 1);
-            taskToComplete.markAsDone();
-            String doneMessage = "Nice! I've marked this task as done:\n" + "  " + taskToComplete;
-            replyFormat(doneMessage);
-        } else {
-            throw new DukeException("Task " + taskNo + " is not in the task list!");
+    private static void completeTask(String promptDescription) throws DukeException {
+        try {
+            int taskNo = Integer.valueOf(promptDescription);
+            if (taskNo > 0 && taskNo <= tasksList.size()) {
+                Task taskToComplete = tasksList.get(taskNo - 1);
+                taskToComplete.markAsDone();
+                String doneMessage = "Nice! I've marked this task as done:\n" + "  " + taskToComplete;
+                replyFormat(doneMessage);
+            } else {
+                throw new DukeException("Task " + taskNo + " is not in the task list!");
+            }
+        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! Please specify a task number. (E.g. done 2)");
+        }
+    }
+
+    // Deletes a task from the taskList
+    private static void deleteTask(String promptDescription) throws DukeException {
+        try {
+            int taskNo = Integer.valueOf(promptDescription);
+            if (taskNo > 0 && taskNo <= tasksList.size()) {
+                counter--;
+                Task taskToDelete = tasksList.get(taskNo - 1);
+                String deletedMessage = "Noted. I've removed this task:\n" + "  " + taskToDelete + "\n"
```

Comment:
Perhaps "deletedMessage" can be changed into "deletedTaskMessage"? Just so that others won't get confused if the deletedMessage variable appears later and they are confused that it refers to a deleted message.